cli/pkg/healthcheck/healthcheck.go

```diff
@@ -141,46 +143,86 @@ func (hc *HealthChecker) globalChecks() []healthcheck.Checker {
 }
 
 func (hc *HealthChecker) deploymentChecks(name string) []healthcheck.Checker {
-	var deploy *appsv1.Deployment
-	var pod v1.Pod
+	getWorkloadLabels := func(ctx context.Context) (map[string]string, error) {
+		deploy, err := hc.k8s.Deployment(ctx, name)
+		if err != nil {
+			return nil, err
+		}
+		return deploy.GetLabels(), nil
+	}
```

Comment:
It is a bit of a pitty there is not a more generic way to handle this

Reply:
yeah agree, i went through a similar exercise. one way this could be more generic would be to add a new method to the `k8s.Client` interface:
```go
Client interface {
	Workload(ctx context.Context, kind string, name string) (runtime.Object, error)
}
```
... but that seemed like overkill, so i left it as-is.